docs/products/access/api-reference/custom-registration/custom-registration-v1.md

```diff
@@ -0,0 +1,149 @@
+| `transaction_id` | Generated in **Init step**. For TWO_STEP, ensures same transaction                                       |
+| `data`           | Raw response coming from the script engine                                                               |
+| `status`         | Status indicating whether the request was successful. See [status codes](#extension-engine-status-codes) |
+
+## Complete Step
+
+Endpoint: `POST /oauth/custom-registration/{idp}/complete`
+
+| Parameter | Description                  |
+|-----------|------------------------------|
+| `idp`     | Identity provider identifier |
+
+JSON body parameters:
+
+| Param            | Required                                | Description                                                                        |
+|------------------|-----------------------------------------|------------------------------------------------------------------------------------|
+| `transaction_id` | yes (`TWO_STEP`) otherwise optional     | Generated in **Init step**. For `TWO_STEP`, ensures same transaction               |
+| `data`           | no                                      | Raw registration request data which will be provided to the Extension Engine       |
+| `scope`          | no                                      | An array of scopes. If none are specified the default scopes are granted.          |
+
+Example request:
+
+```http
+POST /oauth/custom-registration/example-custom-registration-idp/complete HTTP/1.1
+Host: onegini.example.com
+Content-Type: application/json
+Authentication: Basic Y2xpZW50OnNlY3JldA==
+
+{
+  "transaction_id": "123123",
+  "data": "{\"custom_json_key\":\"custom_ json data\"}",      //optional, e.g. challenge code response
+  "scope": ["read", "write"]
+}
+```
+
+Example successful response
+
+```http
+HTTP/1.1 200 Ok
+Content-Type: application/json;charset=UTF-8
+Cache-Control: no-store
+Pragma: no-cache
+
+{
+    "status": 2000,
+    "oauth_token": {
+        "token_type": "bearer",
+        "access_token": "8A5AB83A3C6B7AAC41471C1205167A35E0F9281ED277EE2FDE6E8DE30972936D",
+        "refresh_token": "8CAE26B2B8E8EC18B4D432886448C7F99B558063C517BA41F30966B37C104983",
+        "id_token": "eyJraWQiOiI1Nzk1[...omitted for brevity...]r9KM8c5y-Utpw",
+        "expires_in": 3600
+    },
+    "data": "{\"custom_json_key\":\"custom json data\"}" // optional
+}
+```
+
+In the event of an error in the Access Service, one of the following [error codes](#server-error-codes) will be returned.
+
+Additional error code for this endpoint:
+
+## Server error codes
+
+| Status code | Error code      | Description                                           |
+|-------------|-----------------|-------------------------------------------------------|
+| 400         | invalid_scope   | The requested scope is invalid, unknown or malformed. |
+| 400         | invalid_profile | The specified profile is invalid.                     |
```

Comment:
How can we get an `invalid_profile` error if we do not describe the `profile` tag?

Reply:
I have removed it